spec/barbeque/executor/docker_spec.rb

```diff
@@ -375,6 +413,59 @@
+          let(:container_id2) { '40fa4fcb316f90f65d70717f66357e065747f8c216f392817bb8237f51dc416e' }
+          let(:job_retry2) { FactoryBot.create(:job_retry, job_execution: job_execution, status: :pending) }
+
+          before do
+            container_info['State']['ExitCode'] = 1
+            job_execution.job_definition.update!(slack_notification: slack_notification)
+            allow(Barbeque::SlackClient).to receive(:new).with(slack_notification.channel).and_return(slack_client)
+            FactoryBot.create(:retry_config, job_definition: job_definition, retry_limit: 2)
+          end
+
+          context 'with retry_config and slack_notification (notify_failure_only_if_retry_limit_reached: false)' do
+            let(:slack_notification) { FactoryBot.create(:slack_notification, notify_success: false ) }
+
+            it 'performs retry with sending slack notification for two times' do
+              expect(Open3).to receive(:capture3).with('docker', 'logs', container_id).and_return([stdout, stderr, log_status])
+              expect(Barbeque::ExecutionLog).to receive(:save_stdout_and_stderr).with(job_retry, stdout, stderr)
+              expect(Barbeque::MessageRetryingService.sqs_client).to receive(:send_message).with(queue_url: a_kind_of(String), message_body: a_kind_of(String), delay_seconds: a_kind_of(Integer))
+              expect(slack_client).to receive(:notify_failure).twice
+              executor.poll_retry(job_retry)
+
+              Barbeque::DockerContainer.create!(message_id: job_retry2.message_id, container_id: container_id2)
+              expect(Open3).to receive(:capture3).with('docker', 'logs', container_id2).and_return([stdout, stderr, log_status])
+              expect(Barbeque::ExecutionLog).to receive(:save_stdout_and_stderr).with(job_retry2, stdout, stderr)
+              expect(Open3).to receive(:capture3).with('docker', 'inspect', container_id2) {
+                [JSON.dump([container_info]), '', inspect_status]
+              }
+              executor.poll_retry(job_retry2)
+            end
+          end
+
+          context 'with retry_config and slack_notification (notify_failure_only_if_retry_limit_reached: true)' do
+            let(:slack_notification) { FactoryBot.create(:slack_notification, notify_success: false, notify_failure_only_if_retry_limit_reached: true) }
+
+            it 'performs retry with sending only one slack notification' do
+              expect(Open3).to receive(:capture3).with('docker', 'logs', container_id).and_return([stdout, stderr, log_status])
+              expect(Barbeque::ExecutionLog).to receive(:save_stdout_and_stderr).with(job_retry, stdout, stderr)
+              expect(Barbeque::MessageRetryingService.sqs_client).to receive(:send_message).with(queue_url: a_kind_of(String), message_body: a_kind_of(String), delay_seconds: a_kind_of(Integer))
+              expect(slack_client).to receive(:notify_failure).once
```

Comment:
This line should be moved to `executor.poll_retry(job_retry2)` to verify that `notify_failure` is called once from job_retry2

Reply:
Fixed https://github.com/cookpad/barbeque/pull/77/commits/afcfc4a0838221a0dadb3829b6962396a8759d3f